Remotion/Development/UnitTesting.IsolatedCodeRunner/IsolatedCodeRunner.cs

```diff
@@ -0,0 +1,167 @@
+
+      var startInfo = new ProcessStartInfo
+                      {
+                          WindowStyle = ProcessWindowStyle.Hidden,
+                          CreateNoWindow = false,
+                          RedirectStandardOutput = true
+                      };
+
+      startInfo.FileName = appProcessRunnerExe;
+      startInfo.Arguments = string.Join(" ", args.Select(e => $@"""{e.Replace("\"", "\"\"")}"""));
+      startInfo.WorkingDirectory = Path.GetDirectoryName(_targetMethod.DeclaringType!.Assembly.Location);
+
+      startInfo.Environment[TargetAssemblyPath] = _assemblyLocation;
+      startInfo.Environment[TargetTypeName] = _typeName;
+      startInfo.Environment[TargetMethodName] = _methodName;
+
+      startInfo.Environment[TargetBaseDirectory] = AppContext.BaseDirectory;
+
+      if (!string.IsNullOrEmpty(ConfigFile))
+        startInfo.Environment[TargetConfigFile] = ConfigFile;
+
+      using var process = new Process();
+      process.StartInfo = startInfo;
+      process.OutputDataReceived += DataReceived;
+
+      process.Start();
+      process.BeginOutputReadLine();
+
+      if (process.WaitForExit((int)_timeout.TotalMilliseconds))
+      {
+        // Need to call .WaitForExit() again to make sure that standard out is flushed
+        process.WaitForExit();
+      }
+      else
+      {
+        process.Kill();
+        throw new InvalidOperationException("Isolated code runner execution failed to stop within the specified timeout.");
+      }
+
+      if (_exception != null)
+        throw _exception;
+
+      if (process.ExitCode != 0)
+        throw new InvalidOperationException($"Isolated code runner execution failed with exit code {process.ExitCode}.");
+    }
+
+    private void DataReceived (object sender, DataReceivedEventArgs args)
+    {
+      var line = args.Data;
+      if (line == null)
+        return;
+
+      Console.WriteLine(line);
+      if (line.StartsWith(ExceptionMarker))
+      {
+        // Serialization format (keep in sync with Program.cs):
+        // ##exception:<base64 exception type name>;<base 64 exception message>;<base 64 exception stack trace>
+        var parts = line[ExceptionMarker.Length..].Split(';');
+
+        var exceptionTypeName = DecodeBase64(parts[0]);
+        var exceptionMessage = DecodeBase64(parts[1]);
+        var exceptionStackTrace = DecodeBase64(parts[2]);
+
+        // Try to recreate the original exception in this context
+        Exception? remoteException = null;
+        try
+        {
+          remoteException = (Exception?)Activator.CreateInstance(Type.GetType(exceptionTypeName)!, exceptionMessage);
+        }
+        catch
+        {
+          try
+          {
+            remoteException = (Exception?)Activator.CreateInstance(Type.GetType(exceptionTypeName)!, exceptionMessage, null);
+          }
+          catch
+          {
+            // Ignore and use a IsolatedCodeException instead
+          }
+        }
```

Comment:
I understand what you're doing, it just feels...like we could do this with more typesafety by getting the ctor and failing/fallback to next signatur.